mne/io/boxy/boxy.py

```diff
@@ -234,27 +295,32 @@ def __init__(self, fname, preload=False, verbose=None):
                                     fiducial_coords[1], 
                                     fiducial_coords[2])
         
-        for i_chan in range(len(boxy_labels)-1):
+        for i_chan in range(len(boxy_labels)):
             temp_ch_src_det = apply_trans(native_head_t, boxy_coords[i_chan][:9].reshape(3, 3)).ravel()
             temp_other = np.asarray(boxy_coords[i_chan][9:], dtype=np.float64) # add wavelength and placeholders
-            info['chs'][i_chan]['loc'] = np.concatenate((temp_ch_src_det, temp_other), axis=0)
-        info['chs'][-1]['loc'] = np.zeros((12,))   #remove last line?     
-        
+            info['chs'][i_chan]['loc'] = np.concatenate((temp_ch_src_det, temp_other), axis=0) 
+                
         raw_extras = {'source_num': source_num,
                      'detect_num': detect_num, 
                      'start_line': start_line,
                      'end_line': end_line,
                      'filetype': filetype,
-                     'files': files,}
-
-        print('Start Line: ', start_line)
-        print('End Line: ', end_line)
-        print('Original Difference: ', end_line-start_line)
-        first_samps = start_line
+                     'files': files,
+                     'data_types': data_types,}
+        
+        print('Start Line: ', start_line[0])
+        print('End Line: ', end_line[0])
+        print('Original Difference: ', end_line[0]-start_line[0])
+        first_samps = start_line[0]
         print('New first_samps: ', first_samps)
-        diff = end_line-start_line
+        diff = end_line[0] - start_line[0]
```

Comment:
so do these only test the first montage of n? should probably loop through montages and test each one for expected length, etc.

Reply:
yeah I initially checked each montage but I was getting errors when I passed multiple values to first_samps and last_samps.

The montages had the same amount of data, even though start and end line would differ. This is probably because you can tell boxy to record only so many samples per file but perhaps we should add a check for that in case that setting isn't used.

Reply:
just want to check that each data file the data is the same size as we expect it to be , if the Data objects gets made to big or small we get an error (or maybe only if its too big but not too small)

Reply:
okay sounds good. too big is probably more of a problem since it seems to break plotting.

but yeah I'll throw an error if the file size differ, or if the data will be too big.